selfdrive/sensord/tests/test_sensord.py

```diff
@@ -79,30 +80,49 @@
 }
 
 SENSOR_BUS = 1
-I2C_ADDR_LSM = 0x6A
+I2C_ADDR_BMX_ACCEL = 0x18
+BMX_REG_ID = 0x00
 LSM_INT_GPIO = 84
 
-def read_sensor_events(duration_sec):
-  sensor_events = messaging.sub_sock("sensorEvents", timeout=0.1)
-  start_time_sec = time.monotonic()
-  events = []
-  while time.monotonic() - start_time_sec < duration_sec:
-    events += messaging.drain_sock(sensor_events)
-    time.sleep(0.01)
-
-  assert len(events) != 0, "No sensor events collected"
-  return events
-
 def get_proc_interrupts(int_pin):
   with open("/proc/interrupts") as f:
     lines = f.read().split("\n")
 
   for line in lines:
     if f" {int_pin} " in line:
       return ''.join(list(filter(lambda e: e != '', line.split(' ')))[1:6])
-
   return ""
 
+def is_bmx_available() -> bool:
+  try:
+    # to check one is enough, fails on if bmx is not present
+    with SMBus(SENSOR_BUS, force=True) as bus:
+      bus.read_byte_data(I2C_ADDR_BMX_ACCEL, BMX_REG_ID)
+    return True
+  except Exception:
+    return False
+
+def read_sensor_events(sensor_types, duration_sec):
+  esocks = {}
+  events = {}
+  for stype in sensor_types:
+    esocks[stype] = messaging.sub_sock(stype, timeout=0.1)
+    events[stype] = []
+
+  start_time_sec = time.monotonic()
+  while time.monotonic() - start_time_sec < duration_sec:
+    for esock in esocks:
+      events[esock] += messaging.drain_sock(esocks[esock])
+    time.sleep(0.01)
```

Comment:
this sleep can be longer, the buffers are quite big
```suggestion
    time.sleep(0.1)
```

Reply:
done